doc/task_breakdown.md

```diff
@@ -0,0 +1,70 @@
+# Task Breakdown for SQL Query Builder Development
+
+## 1. Core Modules of the Program
+
+### 1.1 Table Management Module
+- Tasks:
+  - Implement functionality to display a list of tables from the database.
+  - Add the ability to select multiple tables simultaneously.
+  - Configure the removal of tables from the current selection set.
+- Technical Details:
+  - Use an API to fetch the list of tables from the database.
+  - Ensure that removing a table correctly updates associated data (sorting, filters, relationships).
+
+### 1.2 Relationship Management Module
+- Tasks:
+  - Implement automatic detection of relationships between tables (based on foreign keys).
+  - Add the ability to create, modify, and delete relationships manually.
+- Technical Details:
+  - Verify the correct removal of a relationship if one of the tables is deselected.
+  - Add error messages for incorrectly configured relationships.
+
+### 1.3 Column Management Module
+- Tasks:
+  - Display a list of available columns for the selected tables.
```

Comment:
can you try to understand what is the difference between this line and next? just selection?